x-pack/plugins/monitoring/public/components/metricbeat_migration/flyout/flyout.js

```diff
@@ -0,0 +1,341 @@
+          doneWithMigration: onClose,
+          esMonitoringUrl,
+          checkForMigrationStatus: this.checkForMigrationStatus,
+          checkingMigrationStatus,
+          hasCheckedStatus: checkedStatusByStep[activeStep],
+          autoCheckIntervalInMs: AUTO_CHECK_INTERVAL_IN_MS,
+        });
+
+        let calloutTitle = null;
+        if (productName === KIBANA_SYSTEM_ID) {
+          calloutTitle = (
+            <FormattedMessage
+              id="xpack.monitoring.metricbeatMigration.flyout.kibana.migrationDescriptionLabel"
+              defaultMessage="Migrating instance {instance}"
+              values={{
+                instance: (<Monospace>{instance.name}</Monospace>)
+              }}
+            />
+          );
+        }
+        else if (productName === ELASTICSEARCH_CUSTOM_ID) {
+          if (instance) {
+            calloutTitle = (
+              <FormattedMessage
+                id="xpack.monitoring.metricbeatMigration.flyout.elasticsearch.migrationDescriptionLabel"
+                defaultMessage="Migrating node {instance}"
+                values={{
+                  instance: (<Monospace>{instance.name}</Monospace>)
+                }}
+              />
+            );
+          }
+          else {
+            calloutTitle = (
+              <FormattedMessage
+                id="xpack.monitoring.metricbeatMigration.flyout.elasticsearch.migrationDescriptionNoInstanceLabel"
+                defaultMessage="Migrating Elasticsearch nodes"
+              />
+            );
+          }
+        }
+
+        return (
+          <Fragment>
+            <EuiCallOut
+              size="s"
+              title={calloutTitle}
+            />
+            <EuiSpacer size="l"/>
+            <EuiSteps steps={instructionSteps}/>
+          </Fragment>
+        );
+    }
+
+    return null;
+  }
+
+  renderActiveStepNextButton() {
+    const { product, productName } = this.props;
+    const { activeStep, esMonitoringUrl } = this.state;
+
+    // It is possible that, during the migration steps, products are not reporting
+    // monitoring data for a period of time outside the window of our server-side check
+    // and this is most likely temporary so we want to be defensive and not error out
+    // and hopefully wait for the next check and this state will be self-corrected.
+    if (!product) {
+      return null;
+    }
+
+    let willDisableDoneButton = !product.isFullyMigrated;
+    let willShowNextButton = activeStep !== INSTRUCTION_STEP_DISABLE_INTERNAL;
+
+    if (activeStep === INSTRUCTION_STEP_ENABLE_METRICBEAT && productName === ELASTICSEARCH_CUSTOM_ID) {
+      willShowNextButton = false;
+      willDisableDoneButton = !product.isPartiallyMigrated;
+    }
+
+    if (willShowNextButton) {
+      let isDisabled = false;
+      let nextStep = null;
+      if (activeStep === INSTRUCTION_STEP_SET_MONITORING_URL) {
+        isDisabled = esMonitoringUrl.length === 0;
```

Comment:
Is it possible that `esMonitoringUrl` might be undefined? Maybe safeguard it eg: 
`!esMonitoringUrl || !esMonitoringUrl.length`